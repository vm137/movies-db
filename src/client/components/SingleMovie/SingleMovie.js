import React, {Component} from 'react';
import SearchResults from '../SearchResults/SearchResults';

import './SingleMovie.scss';

export default class SingleMovie extends Component {

    constructor(props) {
        super(props);
    }

    render() {
        return (
          <div>
            <h2>Single Movie.</h2>

            <SearchResults />

          </div>
        );
    }
}
