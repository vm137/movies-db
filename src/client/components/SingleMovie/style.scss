@import '../constants/constants';

.single-movie-wrapper {
  height: 480px;
  position: relative;

  &::after {
    content: "";
    background: url('../../img/cinema_dark.jpg');
    background-size: cover;
    opacity: 0.9;
    top: 0;
    left: 0;
    bottom: 0;
    right: 0;
    position: absolute;
    z-index: -1;
  }

  img {
    position: absolute;
    width: 250px;
    top: 50px;
    left: 46px;
    box-shadow: 2px 2px 3px rgba(0, 0, 0, 0.5);
  }

  .btn-search {
    position: absolute;
    top: 17px;
    right: 50px;
    border: 0;
    color: #000;
    width: 90px;
    height: 26px;
    border-radius: 4px;
    outline: none;
    font-size: 12px;
    font-family: Arial, "Helvetica Neue", Helvetica, sans-serif;
    text-transform: uppercase;
    cursor: pointer;

    &:hover {
      background-color: #ff866c;
    }
    &:active {
      background-color: #476a80;
    }
  }

  .movie-details {
    position: absolute;
    top: 48px;
    left: 344px;
    width: 510px;
    color: #fff;
    font-size: 16px;
    font-family: Arial, "Helvetica Neue", Helvetica, sans-serif;

    .title {
      color: $NETFLIX_RED;
      margin: 0 0 6px;
      font-size: 28px;
      text-shadow: 1px 1px 2px rgb(0, 0, 0);
    }

    .tagline {
      margin-bottom: 10px;
      color: $TEXT_LIGHT_GREY;
    }

    .release-wrapper {
      margin-top: 20px;
    }

    .release-date {
      float: left;
      font-size: 18px;
      font-weight: bold;
    }

    .runtime {
      margin-left: 60px;
      font-size: 18px;
      font-weight: bold;
    }

    .overview {
      margin-top: 20px;
      font-size: 15px;
      line-height: 1.2;
      color: $TEXT_LIGHT_GREY;
    }
  }

}
