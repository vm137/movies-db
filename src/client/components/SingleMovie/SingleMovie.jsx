import React from 'react';
import './SingleMovie.scss';

const SingleMovie = () => <h1>SingleMovie</h1>;

// export default class SingleMovie extends Component {
//   constructor(props) {
//     super(props);
//     this.state = {
//       movieId: null,
//     };
//   }
//
//   render() {
//     return (
//       <div>
//         <h2>Single Movie.</h2>
//
//         {this.props.movie}
//
//         <SearchResults movies={this.props.movies} />
//
//       </div>
//     );
//   }
// }

export default SingleMovie;
