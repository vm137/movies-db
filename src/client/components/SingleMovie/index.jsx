import { withRouter } from 'react-router-dom';
import { connect } from 'react-redux';
import SingleMovie from './SingleMovie';
import { eraseSingleMovieAction, fetchSingleMovieAction } from '../../actions/actions';

const mapStateToProps = state => ({
  movieR: state.movie,
});

const mapDispatchProps = dispatch => ({
  eraseSingleMovieAction: () => dispatch(eraseSingleMovieAction()),
  fetchSingleMovie: movieId => dispatch(fetchSingleMovieAction(movieId)),
});

export default withRouter(connect(mapStateToProps, mapDispatchProps)(SingleMovie));
