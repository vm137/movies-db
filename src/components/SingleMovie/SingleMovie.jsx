// @flow

import React, { PureComponent } from 'react';
import Logo from '../Logo';
import './style.scss';

type DefaultProps = {| history: {}, match: { params: {id: ''} } |};
type Props = {
  ...DefaultProps,
  movieR: Object,
  fetchSingleMovie: Function,
  eraseSingleMovieAction: Function,
  history: Object,
  match: Object,
};

export default class SingleMovie extends PureComponent<Props> {
  constructor(props: Props) {
    super(props);
    (this: Object).handleClick = this.handleClick.bind(this);
  }

  componentDidMount() {
    const { fetchSingleMovie, match } = this.props;
    const movieId = match.params.id;
    fetchSingleMovie(movieId);
  }

  handleClick() {
    const { history, eraseSingleMovieAction } = this.props;
    eraseSingleMovieAction();
    history.push('/search');
  }

  render() {
    const { movieR } = this.props;

    return (
      <div className="single-movie-wrapper">
        <Logo />
        <img src={movieR.poster_path} alt={movieR.title} />
        <button className="btn-search" type="button" onClick={this.handleClick}>back</button>

        <div className="movie-details">
          <h2 className="title">{movieR.title}</h2>
          <p className="tagline">{movieR.tagline}</p>
          <div className="release-wrapper">
            <p className="release-date">{movieR.release_date && (movieR.release_date).substr(0, 4)}</p>
            <p className="runtime">
              {movieR.runtime}
&nbsp;min
            </p>
          </div>
          <p className="overview">{movieR.overview}</p>
        </div>

      </div>
    );
  }
}
