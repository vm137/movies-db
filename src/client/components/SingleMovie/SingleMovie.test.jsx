import { render } from 'enzyme';
import React from 'react';
import SingleMovie from './SingleMovie';

const movieId = 10;
const onClick = () => {};

describe('SingleMovie', () => {
  it('renders matching snapshot', () => {
    const wrapper = render(<SingleMovie
      movieId={movieId}
      onClick={onClick}
    />);
    expect(wrapper).toMatchSnapshot();
  });

  // it('mount two', () => {
  //   const wrapper = mount(<SingleMovie
  //     movieId={movieId}
  //     onClick={onClick}
  //   />);
  //   expect(wrapper.find('.single-movie-wrapper').length).toBe(1);
  // });
});
