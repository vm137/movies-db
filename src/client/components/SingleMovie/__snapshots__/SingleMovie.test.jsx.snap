// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SingleMovie /> renders matching snapshot 1`] = `
<div
  className="single-movie-wrapper"
>
  <Logo />
  <img
    alt="Duck, You Sucker"
    src="https://image.tmdb.org/t/p/w500/nQkr1yi87pX1d3lQVTrBEWlCpT1.jpg"
  />
  <button
    className="btn-search"
    onClick={[Function]}
    type="button"
  >
    search
  </button>
  <div
    className="movie-details"
  >
    <h2
      className="title"
    >
      Duck, You Sucker
    </h2>
    <p
      className="tagline"
    >
      Rod Steiger and James Coburn will blow you apart in
    </p>
    <div
      className="release-wrapper"
    >
      <p
        className="release-date"
      >
        1971
      </p>
      <p
        className="runtime"
      >
        157
        Â min
      </p>
    </div>
    <p
      className="overview"
    >
      At the beginning of the 1913 Mexican Revolution, greedy bandit Juan Miranda and idealist John H. Mallory, an Irish Republican Army explosives expert on the lam from the British, fall in with a band of revolutionaries plotting to strike a national bank. When it turns out that the government has been using the bank as a hiding place for illegally detained political prisoners -- who are freed by the blast -- Miranda becomes a revolutionary hero against his will.
    </p>
  </div>
</div>
`;
